homeassistant/components/template/config_flow.py

```diff
@@ -0,0 +1,320 @@
+"""Config flow for the Template integration."""
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine, Mapping
+from typing import Any, cast
+
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.components.sensor import (
+    CONF_STATE_CLASS,
+    DEVICE_CLASS_UNITS,
+    SensorDeviceClass,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    CONF_DEVICE_CLASS,
+    CONF_NAME,
+    CONF_STATE,
+    CONF_UNIT_OF_MEASUREMENT,
+    Platform,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import selector
+from homeassistant.helpers.schema_config_entry_flow import (
+    SchemaCommonFlowHandler,
+    SchemaConfigFlowHandler,
+    SchemaFlowFormStep,
+    SchemaFlowMenuStep,
+)
+
+from .const import DOMAIN
+from .sensor import async_create_preview_sensor
+from .template_entity import TemplateEntity
+
+NONE_SENTINEL = "none"
+
+
+def generate_schema(domain: str) -> dict[vol.Marker, Any]:
+    """Generate schema."""
+    schema: dict[vol.Marker, Any] = {}
+
+    if domain == Platform.SENSOR:
+        schema = {
+            vol.Optional(
+                CONF_UNIT_OF_MEASUREMENT, default=NONE_SENTINEL
+            ): selector.SelectSelector(
+                selector.SelectSelectorConfig(
+                    options=[
+                        "none",
```

Comment:
Why don't we use the `NONE_SENTINEL` here but in the other selectors?

Reply:
Fixed!